frontend/src/components/views/ProjectileView.jsx

```diff
@@ -3,35 +3,69 @@ import { convertToRenderCoordinates, modelFiles } from "../../utils/render_utils
 import { useLoader } from "@react-three/fiber";
 import { Line } from '@react-three/drei';
 
+const renderLaser = (laser, _gltf) => {
+    const { path } = laser;
+    const points = [path[1], path[path.length-1]] // start path at 1 to render laser outside tower
+        .map((point) => {
+            const [x, y, z] = point;
+            return convertToRenderCoordinates({x, y, z});
+        })
+    return (<Line 
+        key={projectileKey(laser)} 
+        points={points} color="red" lineWidth={2} 
+    />)
+}
+
+const renderSpikes = (spikes, gltf) => {
+    const { quaternion, scale } = spikes;
+    const ARRAY_OFFSET = 0.2;
+    const spikesArray = [];
+    for (let x = -ARRAY_OFFSET; x <= ARRAY_OFFSET; x+=ARRAY_OFFSET) {
+        for (let y = -ARRAY_OFFSET; y <= ARRAY_OFFSET; y+=ARRAY_OFFSET) {
+            const coordinates = convertToRenderCoordinates(spikes, {x, y: y - 0.02, z: -0.3});
+            spikesArray.push(coordinates);
+        }
+    }
+
+    return (<group key={projectileKey(spikes)}>
+        {spikesArray.map((coordinates) => {
+            return <primitive
+                key={`spike,${coordinates.x},${coordinates.y},${coordinates.z}`}
+                object={gltf.scene.clone(true)}
+                scale={scale}
+                position={[coordinates.x, coordinates.y, coordinates.z]}
+                quaternion={quaternion}
+            />
+        })}
+    </group>)
+}
+
+const specialProjectiles = {
```

Comment:
Do we have a convention for constants like this already? Perhaps `SPECIAL_PROJECTILES`?